src/PrestaShopBundle/Api/QueryParamsCollection.php

```diff
@@ -114,6 +114,26 @@ public function fromRequest(Request $request)
         return $this;
     }
 
+    /**
+     * @param array $queryParams
+     * @param array $allParams
+     *
+     * @return $this
+     */
+    public function fromArray(array $queryParams, array $allParams = []): QueryParamsCollection
```

Comment:
I have no idea what is this, but Im not gonna dig into it, I believe you :smile:

Reply:
No it's only to get StockMovement so it's only related to the Product stock The existent implementation was very coupled with query so it didn't allow to perform the same query in test environment with simple array input